operator/apis/lifecycle/v1alpha3/keptntaskdefinition_types.go

```diff
@@ -125,3 +132,50 @@ type KeptnTaskDefinitionList struct {
 func init() {
 	SchemeBuilder.Register(&KeptnTaskDefinition{}, &KeptnTaskDefinitionList{})
 }
+
+func (d KeptnTaskDefinition) SpecExists() bool {
+	return d.IsJSSpecDefined() || d.IsContainerSpecDefined()
+}
+
+func (d KeptnTaskDefinition) IsJSSpecDefined() bool {
+	return d.Spec.Function != nil
+}
+
+func (d KeptnTaskDefinition) IsContainerSpecDefined() bool {
+	return d.Spec.Container != nil
+}
+
+func (d KeptnTaskDefinition) IsVolumeMountPresent() bool {
+	return d.IsContainerSpecDefined() && d.Spec.Container.VolumeMounts != nil && len(d.Spec.Container.VolumeMounts) > 0
+}
+
+func (d KeptnTaskDefinition) GetVolumeSource() *v1.EmptyDirVolumeSource {
+	if d.IsContainerSpecDefined() {
```

Comment:
to reduce the nesting of the `if` statements here, I would suggest to check for `!d.isContainerSpecDefined()`  and return the default value in this case, and continue with the other checks afterwards